src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/DataImportService.cs

```diff
@@ -148,5 +151,63 @@ public async Task UpdateStatus(Guid id, DataImportStatus newStatus, double perce
             context.DataImports.Update(import);
             await context.SaveChangesAsync();
         }
+
+        public async Task WriteDataSetFileMeta(Guid subjectId)
+        {
+            await using var contentDbContext = _dbContextSupplier.CreateDbContext<ContentDbContext>();
+            await using var statisticsDbContext = _dbContextSupplier.CreateDbContext<StatisticsDbContext>();
+
+            var observations = await statisticsDbContext.Observation
```

Comment:
As per previous comments, could this potentially be better left unmaterialised (as a Queryable) and therefore used as a subquery to find the Dicstict items further on down in the codebase?  Just wondering about performance for those 6 million+ row Data files if performance could be an issue?

Happy as well to check it out against those big files (QUA01) and see how it fares?